Detectors/FIT/FDD/simulation/src/digit2raw.cxx

```diff
@@ -82,51 +108,54 @@ int main(int argc, char** argv)
     o2::conf::ConfigurableParam::updateFromFile(confDig, "HBFUtils");
   }
   o2::conf::ConfigurableParam::updateFromString(vm["configKeyValues"].as<std::string>());
-  digit2raw(vm["input-file"].as<std::string>(),
-            vm["output-dir"].as<std::string>(),
-            vm["verbosity"].as<int>(),
-            vm["file-for"].as<std::string>(),
-            vm["rdh-version"].as<uint32_t>(),
-            vm["no-empty-hbf"].as<bool>(),
-            vm["flp-name"].as<std::string>(),
-            vm["ccdb-path"].as<std::string>(),
-            vm["lut-path"].as<std::string>());
 
+  const Configuration cfg(vm);
+  if (!cfg.mEnablePadding) {
+    digit2raw<o2::fdd::RawWriterFDD>(cfg);
+  } else {
+    digit2raw<o2::fdd::RawWriterFDD_padded>(cfg);
+  }
   o2::raw::HBFUtils::Instance().print();
 
   return 0;
 }
 
-void digit2raw(const std::string& inpName, const std::string& outDir, int verbosity, const std::string& fileFor, uint32_t rdhV, bool noEmptyHBF, const std::string& flpName, const std::string& ccdbUrl, const std::string& lutPath, int superPageSizeInB)
+template <typename RawWriterType>
+void digit2raw(const Configuration& cfg)
 {
   TStopwatch swTot;
   swTot.Start();
-  o2::fdd::RawWriterFDD m2r;
-  m2r.setFileFor(fileFor);
-  m2r.setFlpName(flpName);
-  m2r.setVerbosity(verbosity);
-  if (ccdbUrl != "") {
-    m2r.setCCDBurl(ccdbUrl);
+  RawWriterType m2r;
+  m2r.setFileFor(cfg.mFileFor);
+  m2r.setFlpName(cfg.mFlpName);
+  m2r.setVerbosity(cfg.mVerbosity);
+  if (cfg.mCcdbPath != "") {
+    m2r.setCCDBurl(cfg.mCcdbPath);
   }
-  if (lutPath != "") {
-    m2r.setLUTpath(lutPath);
+  if (cfg.mChannelMapPath != "") {
+    m2r.setLUTpath(cfg.mChannelMapPath);
   }
   auto& wr = m2r.getWriter();
   std::string inputGRP = o2::base::NameConf::getGRPFileName();
   const auto grp = o2::parameters::GRPObject::loadFrom(inputGRP);
   wr.setContinuousReadout(grp->isDetContinuousReadOut(o2::detectors::DetID::FDD)); // must be set explicitly
+  const int superPageSizeInB = 1024 * 1024;
   wr.setSuperPageSize(superPageSizeInB);
-  wr.useRDHVersion(rdhV);
-  wr.setDontFillEmptyHBF(noEmptyHBF);
-
-  o2::raw::assertOutputDirectory(outDir);
+  wr.useRDHVersion(cfg.mRdhVersion);
+  wr.setDontFillEmptyHBF(cfg.mNoEmptyHBF);
+  wr.useRDHDataFormat(cfg.mDataFormat);
+  if (!cfg.mEnablePadding) { // CRU page alignment padding is used only if no GBT word padding is used
+    wr.setAlignmentSize(16); // change to constexpr static field from class?
+    wr.setAlignmentPaddingFiller(0xff);
+  }
```

Comment:
I see this only for the FDD, why not for FT0 and FV0 also?

Reply:
I'm sorry, my bad, I had to merge these commits. I added these options after rebasing with your last RDH 7 update. Here https://github.com/AliceO2Group/AliceO2/blob/cc3ed933c2aa4a25da6972ac01c9153160a76811/Detectors/FIT/FV0/simulation/src/digit2raw.cxx#L147-L150 and https://github.com/AliceO2Group/AliceO2/blob/cc3ed933c2aa4a25da6972ac01c9153160a76811/Detectors/FIT/FT0/simulation/src/digit2raw.cxx#L147-L150

Reply:
ok, sorry, did not notice it was already added.